AAUnlimited/Functions/Notifications.cpp

```diff
@@ -0,0 +1,231 @@
+#include "StdAfx.h"
+
+namespace Notifications {
+
+	bool enabled = false;
+	std::wstring text;
+	std::list<std::tuple< std::wstring, int >> lines;
+	const int fontLayersCount = 9;
+	int outlineLayersCount = 8;		// Count Outline (black) layers to show for each Notifs lines (Depends outline quality)
+	RECT rect[fontLayersCount] = {	// 0 - shadow position | 0-7 - outline font pos | 8 - main font pos
+		{ 1, 41, 1024, 256 },{ -1, 41, 1024, 256 },
+	{ -1, 39, 1024, 256 },{ 1, 39, 1024, 256 },
+	{ -1, 40, 1024, 256 },{ 1, 40, 1024, 256 },
+	{ 0, 39, 1024, 256 },{ 0, 41, 1024, 256 },
+	{ 0, 40, 1024, 256 }
+	};
+	// 0 - outline/shadow | 1 - normal notify | 2 - important
+	D3DCOLOR colors[3] = { D3DCOLOR_RGBA(0, 0, 0, 255),
+		D3DCOLOR_RGBA(255, 190, 100, 255), D3DCOLOR_RGBA(255, 70, 25, 255) };
+
+	DWORD lastPopTime = 0;
+	int duration = 10000;
+	int maxLines = 4; // 4 for this var == 5 in game
+	const char *fontFamily = "Arial";
+	int fontSize = 18;
+	int lineHeight = 24;
+	IUnknown *Font;
+	int areaPosPercentsX = 0; // in `% * 100`
+	int areaPosPercentsY = 0;
+	int outlineSpread = 0;
+	int gameWindowWidth = 0;
+	int gameWindowHeight = 0;
+	int gameWindowMarginY = 0;
+	DWORD notifyCentered = 0;
+	bool separateColorType = true;
+
+	void AddNotification(const char *text, int type_id) {
```

Comment:
Use C++ objects like std::wstring and std::string where you can. Avoid using C strings.
You should switch to wstrings in most cases.

Reply:
Ok